src/components/Home.jsx

```diff
@@ -1,19 +1,97 @@
-import React from 'react';
+import React, { useState } from 'react';
 import getToken from '../lib/tokens';
 import { useHistory } from 'react-router-dom';
+import db from '../lib/firebase';
+import firebase from 'firebase/app';
+
+import './Home.css';
 
 const Home = () => {
   let history = useHistory();
+  const [tokenName, setTokenName] = useState('');
+  const [notification, setNotification] = useState(null);
 
   const generateToken = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+
+    db.collection('tokens')
+      .doc('tokenDocument')
+      .update({
+        tokenArray: firebase.firestore.FieldValue.arrayUnion(token),
+      })
+      .catch((error) => {
+        db.collection('tokens')
+          .doc('tokenDocument')
+          .set({
+            tokenArray: [token],
+          });
+      });
+
     history.push('/list-view');
   };
 
+  const resetError = () => {
+    setTimeout(() => {
+      setNotification(null);
+    }, 10000);
+  };
+
+  const compareToken = (e) => {
+    e.preventDefault();
+
+    db.collection('tokens')
+      .where('tokenArray', 'array-contains', tokenName)
+      .get()
+      .then((querySnapshot) => {
+        let tokenInDB;
+        querySnapshot.forEach((doc) => {
+          tokenInDB = doc.exists;
+        });
```

Comment:
[The docs](https://firebase.google.com/docs/firestore/query-data/queries#array_membership) suggest that:
> If the array has multiple instances of the value you query on, the document is included in the results only once.

Which tells me that we probably don't need to loop over it but can just directly access `querySnapshot[0].doc.exists` (if `doc` is the value between). For added safety, can check the length of querySnapshot before accessing to prevent an index-out-of-bounds type error.

This seems to be working though, so feel free to ignore this, I haven't tested any of this.

Reply:
I thought the same thing! Me and Claire tried this and it broke everything haha 😅 If I remember, the error said that there was no "doc" property of querySnapshot[0, but when we do a forEach loop we can access the doc. 

It's a shame because I think it would look much clearer what we are doing in this bit, but hey ho 🤷🏻‍♀️

Reply:
Just played with it a bit, what about using the `.empty` field on `querySnapshot`? That is, instead of looping and doing all these extra checks, we can be reasonably sure that one was found based on the query written if we use:

```js
if (!querySnapshot.empty) {
  // token was found...
}